src/Framework/Framework/DotVVM.Framework.csproj

```
@@ -124,8 +124,8 @@
     <TypescriptFile Include="Resources/Scripts/**/*.ts" />
   </ItemGroup>
 
-  <!-- DispatchToInnerBuilds is there to run it just once for all TargetFrameworks -->
-  <Target Name="CompileJS" Inputs="@(TypescriptFile)" Outputs="obj/javascript/root-only/dotvvm-root.js;obj/javascript/root-only-debug/dotvvm-root.js;obj/javascript/root-spa/dotvvm-root.js;obj/javascript/root-spa-debug/dotvvm-root.js" BeforeTargets="DispatchToInnerBuilds">
+  <!-- BeforeBuild is ran for every target framework. However, unless its input files change, this target is skipped. -->
+  <Target Name="CompileJS" Inputs="@(TypescriptFile)" Outputs="obj/javascript/root-only/dotvvm-root.js;obj/javascript/root-only-debug/dotvvm-root.js;obj/javascript/root-spa/dotvvm-root.js;obj/javascript/root-spa-debug/dotvvm-root.js" BeforeTargets="BeforeBuild">
```

```suggestion
  <Target Name="CompileJS" Inputs="@(TypescriptFile)" Outputs="obj/javascript/root-only/dotvvm-root.js;obj/javascript/root-only-debug/dotvvm-root.js;obj/javascript/root-spa/dotvvm-root.js;obj/javascript/root-spa-debug/dotvvm-root.js" BeforeTargets="DispatchToInnerBuilds;BeforeBuild">
```


Actually, I just figured out there can be both of them and it works :D

* when building for multiple targets, it is executed only once
* when building only for one target, it also gets executed

---

Seems a little redundant to me, since when building with `DispatchToInnerBuilds;BeforeBuild` and both framework targets then `CompileJS` is hit 3 times (skipped twice), while with just `BeforeBuild` it is hit just twice (and skipped once).

Either way we have to depend on MSBuild to make the right decision and run `CompileJS` just the first time it is hit if ever.

---

For me, this executes the JS build twice in parallel... with both targets, it only executes it once (as it's skipped twice)

---

When you look at the CI, it also runs twice in parallel. The "Github Actions / dotnet unit tests (windows-latest)" seems to fail for this reason (apparently yarn install is not thread safe or something :sweat_smile: )

---

My bad. Forgot to clean the repo before trying this out. Sorry.

---

No problem, we can just MSBuild for causing all these problems :D